src/csharp/Microsoft.Spark.E2ETest/Hadoop/FileSystemTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using Microsoft.Spark.Hadoop.FS;
+using Microsoft.Spark.Hadoop.Conf;
+using Microsoft.Spark.Sql;
+using Microsoft.Spark.UnitTest.TestUtils;
+using Xunit;
+
+namespace Microsoft.Spark.E2ETest.Hadoop
+{
+    [Collection("Spark E2E Tests")]
+    public class FileSystemTests
+    {
+        private readonly SparkSession _spark;
+
+        public FileSystemTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+        }
+
+        /// <summary>
+        /// Test that methods return the expected signature.
+        /// </summary>
+        [Fact]
+        public void TestSignatures()
+        {
+            using var tempDirectory = new TemporaryDirectory();
+
+            using FileSystem fs = Assert.IsAssignableFrom<FileSystem>(
```

Comment:
you can remove `IsAssignableFrom` here since it's not going to compile if it doesn't return assignable `FileSystem`.

Reply:
Removed.